### PR TITLE
FEAT: fetch-next supports set-words/paths, get/lit-args, objects...

### PR DESCRIPTION
**Pull request by hiiamboris**
Extension to `preprocessor/fetch-next` in the light of :point_up: [February 1, 2020 6:15 PM](https://gitter.im/red/bugs?at=5e3596233aca1e4c5f5a11d4)

### CODE REVIEW

File: utils/preprocessor.r [hunk 1]
```diff
@@ -122,39 +128,88 @@ preprocessor: context [
...
+				case [
+					get-word? :item [set/any 'item get/any to word! item]
+					paren?    :item [set/any 'item do item]
+				]
+				either integer? :item [pick value item][select value :item]
+			]
+			unless find [								;-- select-able types
+				block! paren! path! lit-path! set-path! get-path!
+				object! port! error! map!
+			] type?/word get/any 'value [
```
Comment (dockimbel) [inline utils/preprocessor.r]: You could use a `typeset!` here to make `find` perform in constant time.

**Comment by dockimbel** [comment]
Great work! The changes look good, though the tests on Linux are failing.

File: utils/preprocessor.r (hunk 1 above)
Comment (hiiamboris) [reply utils/preprocessor.r]: Yes, good idea!

**Comment by hiiamboris** [comment]
Thanks for the review @dockimbel ☻
Must be OK now.

**Comment by loziniak** [comment]
Perhaps `image!` tests for Linux could be moved to GTK branch?

**Comment by hiiamboris** [comment]
@loziniak it doesn't matter in this case. I just wanted a path that cannot be `select`-ed. `image/size` was just the 1st thing that came to my mind, by `pair/y` works the same.

**Pull request merged**